test/Transforms/PagerandoWrappers/varargs.ll

```diff
@@ -2,30 +2,86 @@
 
 declare void @llvm.va_start(i8*)
 declare void @llvm.va_end(i8*)
+declare void @llvm.va_copy(i8*, i8*)
 
-define i32 @varags(...) {
-  %ap = alloca i8
-  call void @llvm.va_start(i8* %ap)
-  %ret = va_arg i8* %ap, i32
-  call void @llvm.va_end(i8* %ap)
-  ret i32 %ret
-}
+; CHECK-LABEL: define internal void @user()
+; CHECK-NEXT:    call i32 (...) @varags(i32 13, i32 37)
 
 define internal void @user() {
   call i32 (...) @varags(i32 13, i32 37)
   ret void
 }
 
-; CHECK-LABEL: define internal void @user()
-; CHECK-NEXT:    call i32 (...) @varags(i32 13, i32 37)
-
 ; CHECK-LABEL: define i32 @varags(...)
 ; CHECK-NEXT:    %1 = alloca i8
 ; CHECK-NEXT:    call void @llvm.va_start(i8* %1)
 ; CHECK-NEXT:    %2 = call i32 @"varags$$origva"(i8* %1)
+; CHECK-NEXT:    call void @llvm.va_end(i8* %1)
 ; CHECK-NEXT:    ret i32 %2
 
 ; CHECK-LABEL: define hidden i32 @"varags$$origva"(i8*)
-; CHECK-NEXT:    %ret = va_arg i8* %0, i32
-; CHECK-NEXT:    call void @llvm.va_end(i8* %0)
+; CHECK-NEXT:    %va = alloca i8
+; CHECK-NEXT:    call void @llvm.va_copy(i8* %va, i8* %0)
+; CHECK-NEXT:    %ret = va_arg i8* %va, i32
+; CHECK-NEXT:    call void @llvm.va_end(i8* %va)
 ; CHECK-NEXT:    ret i32 %ret
+
+define i32 @varags(...) {
+  %va = alloca i8
```

Comment:
I don't think this should be `alloca i8`. A more realistic example is at http://llvm.org/docs/LangRef.html#variable-argument-handling-intrinsics.

Reply:
Added an additional test with a realistic `va_list` type to make sure our `alloca`s have the correct type and we don't forget the necessary `bitcast`s.